### PR TITLE
TUF allows optional hashes and length on snapshots

### PR DESCRIPTION
**Pull request by loosebazooka**
This is defined in the spec here:
https://github.com/theupdateframework/specification/blob/53bb11dcd448cc9bd7c657eb3bbd72b073c683e6/tuf-spec.md#file-formats-snapshotjson--file-formats-snapshot

### CODE REVIEW

File: sigstore-java/src/main/java/dev/sigstore/tuf/model/SnapshotMeta.java [hunk 1]
```diff
@@ -29,6 +31,8 @@
 @Value.Immutable
 public interface SnapshotMeta extends TufMeta {
 
+  Integer DEFAULT_MAX_LENGTH = 2000000;
```
Comment (patflynn) [inline sigstore-java/src/main/java/dev/sigstore/tuf/model/SnapshotMeta.java:34]: just curious where the default comes from?

**Comment by patflynn** [comment]
don't we need some test data without the values?

File: sigstore-java/src/main/java/dev/sigstore/tuf/model/SnapshotMeta.java (hunk 1 above)
Comment (loosebazooka) [reply sigstore-java/src/main/java/dev/sigstore/tuf/model/SnapshotMeta.java:34]: oh guess I never added the comment, it comes from the python-tuf defaults.

**Comment by loosebazooka** [comment]
> don't we need some test data without the values?

lol yeah, we definitely do. Been looking through how you generated the old tuf data

**Comment by loosebazooka** [comment]
kk tests update. 

added @jku for sanity check

**Review by jku — APPROVED** [review]
Looks fine to me.

**Review by patflynn — APPROVED** [review]

**Pull request merged**